test/src/node.rs

```diff
@@ -94,6 +96,43 @@ impl Node {
         &self.working_dir
     }
 
+    pub fn log_file(&self) -> String {
+        node_log(&self.working_dir()).display().to_string()
+    }
+
+    pub fn log_monitor(&self, expected: &str, timeout: u64) -> Receiver<String> {
```

Comment:
We can remember the file size when the monitor is created. The monitor only needs to parse the new appended text content (content after the remembered file size). It can save time when the log file is large and we know exactly the event happens after the monitor is created. It also helps when we want to wait on several events in sequence.